ateam_radio_bridge/src/conversion.cpp

```diff
@@ -64,4 +64,75 @@ ateam_msgs::msg::RobotFeedback Convert(const BasicTelemetry & basic_telemetry)
   return robot_feedback;
 }
 
+void ConvertFloatArrayToVec3(const float (&float_arr_3)[3], geometry_msgs::msg::Vector3 & vec) {
+  vec.x = float_arr_3[0];
+  vec.y = float_arr_3[1];
+  vec.z = float_arr_3[2];
+}
+
+geometry_msgs::msg::Vector3 ConvertFloatArrayToVec3(const float (&float_arr_3)[3]) {
+  geometry_msgs::msg::Vector3 vec;
+
+  ConvertFloatArrayToVec3(float_arr_3, vec);
+
+  return vec;
+}
+
+void ConvertFloatArrayToTwist(const float (&fw_state_space_array)[3], geometry_msgs::msg::Twist & twist) {
+  twist.linear.x = fw_state_space_array[0];
+  twist.linear.y = fw_state_space_array[1];
+  twist.linear.z = 0.0;
+  twist.angular.x = 0.0;
+  twist.angular.y = 0.0;
+  twist.angular.z = fw_state_space_array[2];
+}
+
+geometry_msgs::msg::Twist ConvertFloatArrayToTwist(const float (&fw_state_space_array)[3]) {
+  geometry_msgs::msg::Twist twist;
+
+  ConvertFloatArrayToTwist(fw_state_space_array, twist);
+
+  return twist;
+}
```

Comment:
What's the motivation for the extra internal helper that returns void rather than populating the twist directly in the function called by the top-level conversion?

Reply:
probably just thinking too much like C. I guess if all the assignment operators are overloaded there's no reason to have it. I can inline it.